app/controllers/publish/courses/deletions_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Publish
+  module Courses
+    class DeletionsController < PublishController
+      def edit
+        authorize(provider)
+
+        @course_deletion_form = CourseDeletionForm.new(course)
+      end
+
+      def destroy
+        authorize(provider)
+
+        @course_deletion_form = CourseDeletionForm.new(course, params: deletion_params)
+
+        if @course_deletion_form.destroy!
+          flash[:success] = "#{@course.name} (#{@course.course_code}) has been deleted"
+
+          redirect_to publish_provider_recruitment_cycle_courses_path(
+            provider.provider_code,
+            recruitment_cycle.year,
+          )
+        else
+          render :edit
+        end
+      end
+
+    private
+
+      def course
+        @course ||= CourseDecorator.new(provider.courses.find_by!(course_code: params[:code]))
+      end
+
+      def deletion_params
+        return { course_code: nil } if params[:publish_course_deletion_form].blank?
+
+        params
+          .require(:publish_course_deletion_form)
```

Comment:
Shouldn't by virtue of `require` makes it blow up when it is lacking?

Reply:
yup but i don't think we want to blow up every time the user submits incorrectly, we'd rather show the error that some input is required. This is a pattern we used quite a bit in register

Reply:
The http request payload is always there.
![image](https://user-images.githubusercontent.com/470137/160129635-6f915e8d-281d-4478-84f4-190e27fc87fc.png)

Reply:
defo remember that not being there in register lol, have removed anyway